addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js

```diff
@@ -163,7 +163,18 @@ HTMLElement.prototype.oDeleteForward = function (offset) {
         return;
     }
 
-    const nextSibling = this.nextSibling;
+    let nextSibling = this.nextSibling;
+    while (nextSibling && isWhitespace(nextSibling)) {
+        const index = childNodeIndex(nextSibling);
+        const left = getState(nextSibling, index, DIRECTIONS.LEFT).cType;
+        const right = getState(nextSibling, index, DIRECTIONS.RIGHT).cType;
+        if (left === CTYPES.BLOCK_OUTSIDE && right === CTYPES.BLOCK_OUTSIDE) {
+            // If the next sibling is a whitespace, remove it.
+            nextSibling.remove();
+            nextSibling = this.nextSibling;
+        }
+    }
```

Comment:
Saw the changes just now won't this need an `break` in else?

Reply:
I think so, yes. And we'll need a test for it as well because somehow it wasn't caught by any of our current tests.

Reply:
@dmo-odoo oops, I overlooked such case! I have made a new [PR](https://github.com/odoo/odoo/pull/184557) with a test case for it. Please have a look.
@deso-odoo  Thanks for pointing it out.